src/dom.ts

```diff
@@ -286,11 +286,10 @@ export function registerEvListenerAction(elem: EventTarget, add: boolean, event:
       case "mouseover":
       case "mouseout":
       case "mouseleave":
-         let index = hintworthy_js_elems.indexOf(elem)
          if (add) {
-            if (index == -1)
                hintworthy_js_elems.push(elem)
          } else {
+            let index = hintworthy_js_elems.indexOf(elem)
             if (index >= 0)
                hintworthy_js_elems.splice(index, 1)
          }
```

Comment:
Good catch with the multiple removals thing.

So your fix for multiples is to add multiple copies of the element? That's probably fine in practice and I'm happy to accept it like this.

We should note in comments, however, that we could get bugs if people make invalid calls to removeEventListener (e.g. try to remove things that haven't been added). We can avoid this by keeping the full tuple (event type, callback, element, options/useCapture) about in order to accurately replicate what AEL and REL actually do, but it's a little bit fiddly and pretty unlikely that sensible websites will be trying to remove things that aren't there.

Another easier thing we can do for a different class of unlikely errors is putting the addEventListener in a try/catch and only add to the registry if it doesn't fail (it will only fail if you give it primitive values, if you give it an object it assumes that the object supports some interface).